JBPM-7267: Improve computation performance

romartin opened the pull request:
Hey @hasys 

This is the cherry-pick for https://github.com/kiegroup/kie-wb-common/pull/1819 into `master`.
See https://issues.jboss.org/browse/JBPM-7267

Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
@hasys when you have a chance please... it's just the cherry-pick :) Thanks!

romartin commented:
Jenkins please retest this

hasys commented:
Hi @romartin,

it looks like in full downstream build doesn't work keyboard short cuts in IE11 and FF. I tried 7.0.0-CR3 and shortcuts work good there. Not sure is it stops working with this PR or it is not work on master branch in general.

romartin commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/lookup/domain/DomainLookupsCache.java:
```diff
@@ -0,0 +1,110 @@
+        if (null == ids) {
+            return Collections.emptySet();
+        }
+        return ids;
```
hehe good point, well IMO there is no need to use the Optional here, it will create some minor workload, although not really an issue... anyway what about a half-way solution like `return null != ids ? ids : Collections.emptySet()`? :)

romartin commented:
Hey @hasys @jomarko 

Thanks for the review! Added a commit to fix the comment and rebased. 

About the keyboard shortcuts, I've tested now locally and it works for me. Notice I just rebased, so now this stuff is on top of Walter's one as on `7.7.x`, so I guess that could be the issue... anyway let's wait again for the build and let's see... :+1: 

Thanks!

romartin commented:
Jenkins execute full downstream build

romartin commented:
Hey @hasys , sorry as usually you're right...I was just testing Chrome... :) soo in my Linux laptop, the keyboard works on Chrome and FF, but it does not work in the IE case. Just trying to see what's wrong...

romartin commented:
Hey @hasys found the root issue, it's not due to this change. I tested reverting [this PR on master](https://github.com/kiegroup/kie-wb-common/pull/1805), already merged, and now it works fine for me in IE11 too. On the other hand, as you pointed as well, it works on `CR3` because [this another PR](https://github.com/kiegroup/kie-wb-common/pull/1804) was finally NOT merged into `7.7.x`

So just reopened https://issues.jboss.org/browse/JBPM-7132 - @Josephblt can you pelase give it a look?

@hasys please feel free to approve this if you think it's fine.. :+1: 

Thanks guys!

romartin commented:
Jenkins please retest this
